pkg/nar/header.go

```diff
@@ -15,6 +17,44 @@ type Header struct {
 	Executable bool     // Set to true for files that are executable
 }
 
+// Validate does some consistency checking of the header structure, such as
+// checking for valid paths and inconsistent fields, and returns an error if it
+// fails validation.
+func (h *Header) Validate() error {
+	// Path may not start with a /, and may not contain null bytes
+	if len(h.Path) > 1 {
+		if h.Path[:1] == "/" {
+			return fmt.Errorf("path %v starts with a /", h.Path)
+		}
+
+		if strings.ContainsAny(h.Path, "\u0000") {
+			return fmt.Errorf("path contains null bytes")
+		}
+	}
+
+	// Regular files and directories may not have LinkTarget set.
+	if h.Type == TypeRegular || h.Type == TypeDirectory {
+		if h.LinkTarget != "" {
+			if h.LinkTarget != "" {
```

Comment:
I think you can drop one of these checks as they are equivalent

Reply:
Oops, this was some reorganization error. I dropped it here and in ls/list.go.